### DISCUSSION TITLE
snapshot api v1beta1 to v1 migration question

### DISCUSSION THREAD

**Discussion by langjianke** [fabric8io/kubernetes-client#5754] [Q&A]
recently we start to see below errors when we trying to doing snapshot operations

```
Failed to create Volume Snapshot snapshot-22854048-0918-4a94-853f-0fc577b80986-cache, for job id 22854048-0918-4a94-853f-0fc577b80986
io.fabric8.kubernetes.client.KubernetesClientException: Failure executing: POST at: https://34.123.190.81/apis/snapshot.storage.k8s.io/v1beta1/namespaces/build/volumesnapshots. Message: admission webhook "validation-webhook.snapshot.storage.k8s.io" denied the request: expect resource to be {snapshot.storage.k8s.io v1 volumesnapshots}, {snapshot.storage.k8s.io v1 volumesnapshotcontents}, or {snapshot.storage.k8s.io v1 volumesnapshotclasses}, but found {snapshot.storage.k8s.io v1beta1 volumesnapshots}. Received status: Status(apiVersion=v1, code=400, details=null, kind=Status, message=admission webhook "validation-webhook.snapshot.storage.k8s.io" denied the request: expect resource to be {snapshot.storage.k8s.io v1 volumesnapshots}, {snapshot.storage.k8s.io v1 volumesnapshotcontents}, or {snapshot.storage.k8s.io v1 volumesnapshotclasses}, but found {snapshot.storage.k8s.io v1beta1 volumesnapshots}, metadata=ListMeta(_continue=null, remainingItemCount=null, resourceVersion=null, selfLink=null, additionalProperties={}), reason=null, status=Failure, additionalProperties={}).
```
and
```
Uncaught exception
io.fabric8.kubernetes.client.KubernetesClientException: Failure executing: POST at: https://35.232.169.18/apis/snapshot.storage.k8s.io/v1beta1/volumesnapshotclasses. Message: 404 page not found
```

after check around, turns out we should be migrate to v1 api

we are currently on below dependencies

```
        ['kubernetes_client_fabric8', 'io.fabric8:kubernetes-client:4.11.0'],
        ['volume_snapshot_client', 'io.fabric8:volumesnapshot-client:4.11.0'],
        ['volume_snapshot_model', 'io.fabric8:volumesnapshot-model:4.11.0'],
```
I find that in 4.11 volumnsnapshot-model, VolumeSnapshot.java have this line "@Version("v1beat1")", do I need to update above dependencies to newer version to use v1 api? 
such as this 6.X version have this v1 api version : https://github.com/fabric8io/kubernetes-client/blob/cf3b65d636af2bccb5b0833e2399e45115072b9f/extensions/volumesnapshot/model/src/generated/java/io/fabric8/volumesnapshot/api/model/VolumeSnapshot.java#L69

is there a way to keep the older package version and overwrite the api from v1beat1 to v1? I notice there is .withApiVersion() in the snapshotclient call wonder if I can overwrite the api version?

**Comment by rohanKanojia** [comment]
@langjianke : Do you know since which volcano version `v1` apiGroup got introduced? We need to add  `v1` equivalents of [generator-v1beta1](https://github.com/fabric8io/kubernetes-client/tree/main/extensions/volcano/generator-v1beta1) and [model-v1beta1](https://github.com/fabric8io/kubernetes-client/tree/main/extensions/volcano/model-v1beta1)

**Reply by langjianke** [reply]
looks like from io.fabric8:kubernetes-client:5.x.x it becomes v1 (not sure if this answers your question)
I spot check 2 4.x version and 1 5.x, looks like from 5 apiVersion becomes v1
